lib/builtins/deploy-delegates/cfn-deployer/helper.js

```diff
@@ -99,6 +100,9 @@ function deployStack(reporter, options, callback) {
             }
         ]
     };
+
+    _mergeTemplateParameters(stackOptions.stackParameters, templateParameters);
```

Comment:
I'm always thinking that current flow is in need of more transparency regarding the input parameters to CFN stack; skillId and codeBucket/Key/Version are values passed from previous skillMetadata and S3 deployment, only lambdaRuntime and lambdaHandler are the two required values that we read from userConfig, with a special logic to map `userConfig.runtime` and `userConfig.handler` to the two stack parameters. What if we put them as `userConfig.templateParameter.lambdaRuntime` and `userConfig.templateParameter.lambdaHandler`, and use a **$ref** syntax to have a pass-through experiences for stack parameters? Please check this example:

```
 "userConfig": {
    "awsRegion": "us-east-1",
    "templatePath": "./template/stack.yaml"
    "templateParameters": { 
        "LambdaHandler": "nodejs10.x",
        "LambdaRuntime": "index.handler",
        "SkillId": "$state.skillId",
        "CodeBucket": "$state.@ask-cli/cfn-deployer.deployState.s3.bucket",
        "CodeKey": "$state.@ask-cli/cfn-deployer.deployState.s3.key",
        "CodeVersion": "$state.@ask-cli/cfn-deployer.deployState.s3.objectVersion",
        "OtherParameterIfExistInStack": "..."
    }
},
```

In this way, CLI can be a totally pass-through bridge for all the parameters of cfn-stack.

Reply:
@RonWang would this be a breaking change?
When new project is generated, userConfig section looks like
```
"userConfig": {
          "runtime": "nodejs10.x",
          "handler": "index.handler",
          "templatePath": "./infrastructure/cfn-deployer/skill-stack.yaml",
          "awsRegion": "us-east-1"
        }
```
runtime and handler will be moved under templateParameters and renamed. I think this is a breaking change?